packages/storage-s3/src/index.ts

```diff
@@ -32,13 +32,22 @@ export type S3StorageOptions = {
    * Collection options to apply the S3 adapter to.
    */
   collections: Partial<Record<UploadCollectionSlug, Omit<CollectionOptions, 'adapter'> | true>>
+
   /**
    * AWS S3 client configuration. Highly dependent on your AWS setup.
    *
    * [AWS.S3ClientConfig Docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/s3clientconfig.html)
    */
   config: AWS.S3ClientConfig
 
+  /**
+   * Response headers configuration.
+   * Can be used to replace default or append custom headers to the response.
+   *
+   * @default undefined
+   */
+  getResponseHeaders?: (defaultHeaders: HeadersInit) => Headers
```

Comment:
I would make the signature of this function an args object so that it can include the `req`. Also we may as well allow for a `Promise<Headers>`. This will allow for a lot more potential for more use cases that will need this in the future and avoid breaking change.

`getResponseHeaders?: ({ headers, req }: { headers: HeadersInit, req: PayloadRequest }) => Headers | Promise<Headers>`